pkg/event/event.go

```diff
@@ -0,0 +1,86 @@
+package event
+
+import (
+	"context"
+	"sync"
+
+	"github.com/openshift-online/maestro/pkg/api"
+)
+
+// EventClient is a client that can receive resource status change events.
+type EventClient struct {
+	clusterName string
+	resChan     chan *api.Resource
+}
+
+// NewEventClient creates a new event client.
+func NewEventClient(clusterName string) *EventClient {
+	return &EventClient{
+		clusterName: clusterName,
+		resChan:     make(chan *api.Resource),
+	}
+}
+
+// Receive returns a channel that can be used to receive resource status change events.
+func (c *EventClient) Receive() <-chan *api.Resource {
+	return c.resChan
+}
+
+// EventHub is a hub that can broadcast resource status change events to registered clients.
+type EventHub struct {
```

Comment:
is this only used for broadcase? if so EventBroadcaster seems like a better name.

Reply:
updated eventHub to eventBroadcaster.